src/relays/TrustedRelay.sol

```diff
@@ -120,8 +132,45 @@ contract TrustedRelay {
         uint256 expiry,
         uint8 v,
         bytes32 r,
-        bytes32 s
+        bytes32 s,
+        address to,
+        bytes calldata data
     ) external {
+        uint256 startGas = gasleft();
+
+        // Withdraw the L1 DAI to the receiver
+        requestMint(wormholeGUID, signatures, maxFeePercentage, gasFee, expiry, v, r, s);
+
+        // Send the gas fee to the relayer
+        dai.transfer(msg.sender, gasFee);
+
+        // Optionally execute an external call
+        if(to != address(0)) {
+            (bool success,) = to.call(data);
+            if (!success) {
+                // solhint-disable-next-line no-inline-assembly
+                assembly {
+                    returndatacopy(0, 0, returndatasize())
+                    revert(0, returndatasize())
+                }
+            }
+        }
+
+        // If the eth price oracle is enabled, use its value to check that gasFee is within an allowable margin
+        (bytes32 ethPrice, bool ok) = ethPriceOracle.peek();
+        require(!ok || gasFee * RAD <= uint256(ethPrice) * gasMargin * gasprice() * (startGas - gasleft()), "TrustedRelay/excessive-gas-fee");
```

Comment:
Can we reduce the size of this? WAD * RAD sizing makes me nervous of overflow. `gasMargin` should be okay as a WAD or even BPS.

Reply:
Yeah, FWIW using RAY gives us 9 orders of magnitudes of margin before overflowing (log10(2^256)-log10(3000*10^18*2*10^27*1000*10^9*30*10^6) = ~9). But I think you are right, there is no reason not to be on the safe side here. Will use BPS.